src/Functions/FunctionsJSON.h

```diff
@@ -388,21 +395,25 @@ class JSONOverloadResolver : public IFunctionOverloadResolver, WithContext
 
     FunctionBasePtr build(const ColumnsWithTypeAndName & arguments) const override
     {
+        DataTypePtr json_return_type = Impl<DummyJSONParser>::getReturnType(Name::name, createBlockWithNestedColumns(arguments));
```

Comment:
Actually this is not enough to implement this in a proper way (and this is what I was talking about in the #29746)

```sql
-- ok
┌─JSONExtract('["1"]', toNullable(2), 'Nullable(String)')─┐
│ ᴺᵁᴸᴸ                                                    │
└─────────────────────────────────────────────────────────┘

-- not ok
┌─JSONExtractString('["1"]', toNullable(2))─┐
│                                           │
└───────────────────────────────────────────┘
```

So it does not update the Nullable map I guess.

Reply:
Ok, I see. (although it is strange that return value is Nullable, but it does not return NULL when key does not exists, but `JSONExtract('["1"]', toNullable(2), 'String')` has the same semantics anyway)